src/rpcblockchain.cpp

```diff
@@ -632,6 +633,87 @@ Value getchaintips(const Array& params, bool fHelp)
+{
+    if (fHelp || params.size() != 1)
+        throw runtime_error(
+                "getfeeinfo\n"
+                        "\nReturns details of transaction fees over the last n blocks.\n"
+                        "\nArguments:\n"
+                        "1. blocks     (int, required) the number of blocks to get transaction data from\n"
+                        "\nResult:\n"
+                        "{\n"
+                        "  \"txcount\": xxxxx                (numeric) Current tx count\n"
+                        "  \"txbytes\": xxxxx                (numeric) Sum of all tx sizes\n"
+                        "  \"ttlfee\": xxxxx                 (numeric) Sum of all fees\n"
+                        "  \"feeperkb\": xxxxx               (numeric) Average fee per kb over the block range\n"
+                        "  \"rec_highpriorityfee_perkb\": xxxxx    (numeric) Recommended fee per kb to use for a high priority tx\n"
+                        "}\n"
+                        "\nExamples:\n" +
+                HelpExampleCli("getfeeinfo", "5") + HelpExampleRpc("getfeeinfo", "5"));
+
+
+    int nBlocks = params[0].get_int();
+    int nStartHeight = chainActive.Height() - nBlocks;
+    if (nStartHeight <= 0)
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "invalid start height");
+
+    int nBestHeight = chainActive.Height();
+    CAmount nFees = 0;
+    int64_t nBytes = 0;
+    int64_t nTotal = 0;
+    for (int i = nStartHeight; i <= nBestHeight; i++) {
+        CBlockIndex* pindex = chainActive[i];
+        CBlock block;
+        if (!ReadBlockFromDisk(block, pindex))
+            throw JSONRPCError(RPC_DATABASE_ERROR, "failed to read block from disk");
+
+        CAmount nValueIn = 0;
+        CAmount nValueOut = 0;
+        for (const CTransaction& tx : block.vtx) {
+            if (tx.IsCoinBase() || tx.IsCoinStake())
+                continue;
+
+            for (unsigned int j = 0; j < tx.vin.size(); j++) {
+                if (tx.vin[j].scriptSig.IsZerocoinSpend()) {
+                    nValueIn += tx.vin[j].nSequence * COIN;
+                    continue;
+                }
+
+                COutPoint prevout = tx.vin[j].prevout;
+                CTransaction txPrev;
+                uint256 hashBlock;
+                if(!GetTransaction(prevout.hash, txPrev, hashBlock, true))
+                    throw JSONRPCError(RPC_DATABASE_ERROR, "failed to read tx from disk");
+                nValueIn += txPrev.vout[prevout.n].nValue;
+            }
+
+            for (unsigned int j = 0; j < tx.vout.size(); j++) {
+                nValueOut += tx.vout[j].nValue;
+            }
+
+            nFees += nValueIn - nValueOut;
+            nBytes += tx.GetSerializeSize(SER_NETWORK, CLIENT_VERSION);
+            nTotal++;
+        }
+
+        pindex = chainActive.Next(pindex);
+        if (!pindex)
+            break;
+    }
+
+    Object ret;
+    ret.emplace_back(Pair("txcount", (int64_t)nTotal));
```

Comment:
`emblace_back()` isn't supported in the upcoming change to UniValue, use `push_back()` instead.

Reply:
Really? Hmmm thought univalue was supposed to be more inline with current cpp standards.